File: src/main/java/br/ufsm/csi/poow2/papas_freezeria/model/Jogador.java
package br.ufsm.csi.poow2.papas_freezeria.model;

import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serial;
import java.io.Serializable;

@NoArgsConstructor
@AllArgsConstructor
@Entity
@SequenceGenerator(name = "seq_jogador", sequenceName = "seq_jogador", allocationSize = 1)
public class Jogador implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seq_jogador")
    private int id_Jogador;
    @Column
    private String nome;
    @Column
    private String email;
    @Column
    private String senha;
    @Column
    private String permissao;

    public int getIdJogador() {
        return id_Jogador;
    }

    public void setIdJogador(int idJogador) {
        this.id_Jogador = idJogador;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getSenha() {
        return senha;
    }

    public void setSenha(String senha) {
        this.senha = senha;
    }

    public String getPermissao() {
        return permissao;
    }

    public void setPermissao(String permissao) {
        this.permissao = permissao;
    }
}

File: Angular/papas-freezeria/src/app/principal/jogador/jogador.ts
export class Jogador{

  id_Jogador?: string | undefined;
  nome?: string | undefined;
  email?: string | undefined;
  senha?: string | undefined;
  permissao?: string | undefined;
  token?: string | undefined;

  constructor(id_Jogador?: string, nome?: string, email?: string, senha?: string, permissao?: string, token?: string) {
    this.id_Jogador = id_Jogador;
    this.nome = nome;
    this.email = email;
    this.senha = senha;
    this.permissao = permissao;
    this.token = token;
  }

}
